pkg/services/folder/folderimpl/sqlstore.go

```diff
@@ -344,27 +346,74 @@ func (ss *sqlStore) GetHeight(ctx context.Context, foldrUID string, orgID int64,
 	return height, nil
 }
 
-func (ss *sqlStore) GetFolders(ctx context.Context, orgID int64, uids []string) ([]*folder.Folder, error) {
-	if len(uids) == 0 {
+// GetFolders returns org folders by their UIDs.
+// If setFullpath is true it computes also the full path of a folder.
+// The full path is a string that contains the titles of all parent folders separated by a slash.
+// For example, if the folder structure is:
+//
+//	A
+//	└── B
+//	    └── C
+//
+// The full path of C is "A/B/C".
+// The full path of B is "A/B".
+// The full path of A is "A".
+// If a folder contains a slash in its title, it is escaped with a backslash.
+// For example, if the folder structure is:
+//
+//	A
+//	└── B/C
+//
+// The full path of C is "A/B\/C".
+func (ss *sqlStore) GetFolders(ctx context.Context, q folder.GetFoldersQuery) ([]*folder.Folder, error) {
+	if len(q.UIDs) == 0 {
 		return []*folder.Folder{}, nil
 	}
 	var folders []*folder.Folder
 	if err := ss.db.WithDbSession(ctx, func(sess *db.Session) error {
-		b := strings.Builder{}
-		b.WriteString(`SELECT * FROM folder WHERE org_id=? AND uid IN (?` + strings.Repeat(", ?", len(uids)-1) + `)`)
-		args := []any{orgID}
-		for _, uid := range uids {
+		s := `SELECT f0.id, f0.org_id, f0.uid, f0.parent_uid, f0.title, f0.description, f0.created, f0.updated`
+		// compute full path column if requested
+		if q.WithFullpath {
+			s += fmt.Sprintf(`, %s AS fullpath`, getFullpathSQL(ss.db.GetDialect()))
+		}
+		s += ` FROM folder f0`
+		// join the same table multiple times to compute the full path of a folder
+		if q.WithFullpath {
+			s += getFullpathJoinsSQL()
+		}
+		s += ` WHERE f0.org_id=? AND f0.uid IN (?` + strings.Repeat(", ?", len(q.UIDs)-1) + `)`
+		args := []any{q.OrgID}
+		for _, uid := range q.UIDs {
 			args = append(args, uid)
 		}
-		return sess.SQL(b.String(), args...).Find(&folders)
+		return sess.SQL(s, args...).Find(&folders)
```

Comment:
What if the length of the query hits DB limits? UIDs are quite long and, if I am not mistaken, there are no limits for how many folders users can create. 
If we want to keep filtering by a list of UIDs then I think we should do some batching to avoid large queries sent to DB.